### PR TITLE
refact: upgrade to Ant Design 4

### PR DESCRIPTION
**Pull request by davidlougheed**

### CODE REVIEW

**Comment by davidlougheed** [comment]
notif drawer is broken, will do

**Review by gsfk — COMMENTED** [review]
First pass at stuff, haven't hit everything yet:

- I ignored anything that looks like a deprecation warning
- I ignored anything that looks like a redux performance issue, since I'm assuming it's separate. App is getting hard to work with, though.
- I got several "Can't perform a React state update on an unmounted component" errors, but perhaps this is redux again rather than antd?

Broken stuff: 
- The statistics in Experiment overview are jumbled together. 
- the layout of the Dataset edit form is wonky

Different but probably okay:
- drop-box interface shows a smaller drop-down window, at first I wasn't sure if half my drop-box was missing, but the old interface with the huge window is not obviously better

Different but probably not okay:
- The input box in the service request modal is smaller, but the "GET" button is in the same place, so a bit awkward. Either keep the long box or move the button.

**Comment by davidlougheed** [comment]
re: the set state errors, see https://github.com/ant-design/ant-design/issues/42378

I might eliminate the immutability check middleware in dev mode on the app, since it is quite slow.

**Review by gsfk — COMMENTED** [review]
In variants search the "Gene / position" autocomplete menu is now blank, even though it was working during my previous tests (I think... ), and the backend is responding correctly

blank



Otherwise, performance is much better. I still caught a few warnings, but not sure how relevant they are to antd refactor: 

- "Warning: This synthetic event is reused for performance reasons" when using DRS search
- got two "Can't perform a React state update on an unmounted component" warnings, once in variants search, and another in the Tracks tab while igv was waiting to render a vcf,

**Review by SanjeevLakhwani — COMMENTED** [review]
looks fine for now!

File: package-lock.json
[the remark this answers is not in the document]
Comment (davidlougheed) [reply package-lock.json:1]: package lock should always be committed...

**Review by v-rocheleau — APPROVED** [review]
Lgtm! Working as expected

**Pull request merged**